src/main/webapp/app/entities/programming-exercise/instructions/programming-exercise-editable-instruction.component.ts

```diff
@@ -141,6 +145,7 @@ export class ProgrammingExerciseEditableInstructionComponent implements AfterVie
     }
 
     updateProblemStatement(problemStatement: string) {
+        this.editingStarted.emit();
```

Comment:
This will always emit, now only at the first edit :thinking: 
Maybe it should be better named something 'hasUnsavedChanges', so this could be implemented in the setter of `this.unsavedChanges`.
When you subscribe to this event you should probably add `pipe(distinctUntilChanged())` as you are not interested in the same value over and over again.

Reply:
The intention was to allow users to keep their changes after editing the template. But I think it might still be better to always allow them to switch between templates, like you suggested